### ISSUE TITLE
Watcher does not emit any events at all

### ISSUE THREAD

**Issue by YanWittmann** [gmethvin/directory-watcher#96]
Hi there!
I an trying to get your library to work in my application, but whatever I try, I don't seem to get any file creation, modification or deletion events.

I use the exact setup as in your example on the readme:

```java
public static void main(String[] args) throws IOException {
    DirectoryWatcher.builder()
            .path(Path.of("I:", "projects"))
            .listener(event -> System.out.println(event.eventType() + " " + event.path()))
            .build()
            .watch();
}
```

I am on windows (as seen in the file path), and I have tried different methods of passing in the directory I want to watch (e.g. `new File("I:\\projects").toPath()`, etc.)

I am thankful for any help you can give me!

**Comment by gmethvin** [comment]
What's your OS, filesystem, and JDK version? Does this happen for all directories you try to watch, or only on specific drives?

**Comment by YanWittmann** [comment]
Sorry to keep you waiting.

- This happens for every drive I have connected
- I have Windows 10 22H2 19045.4894
- I tested it using Amazon Corretto 21.0.3

... but while testing it using Amazon Corretto 1.8.0 for this issue, it does appear to work?
I can't use that version in my project however, since other dependencies require version 21.

I'm not sure how to continue now.

**Comment by gmethvin** [comment]
There must be something else about your configuration that's causing the behavior. FWIW I created a branch to run against Corretto, JDK 21 and Windows Server 2019 (based on Windows 10) and all the tests pass: https://github.com/gmethvin/directory-watcher/actions/runs/10980872580/job/30487192228

You can try enabling DEBUG logging, e.g.,
https://github.com/gmethvin/directory-watcher/blob/8fc2eaa95e04173f8470eca63864f97b4b614bce/core/src/test/resources/logback-test.xml#L7

That might give you a better idea of what's happening.

**Comment by gmethvin** [comment]
It is interesting that it works on 1.8 but not 21 though. Maybe you can try other JDK versions.

**Comment by YanWittmann** [comment]
Oh my god. I got it now.

```
[DEBUG] [2024-09-22 14:09:19] Registering [C:\Users\yan20\Downloads].
[DEBUG] [2024-09-22 14:09:21] ENTRY_CREATE [C:\Users\yan20\Downloads\New Text Document.txt]
[DEBUG] [2024-09-22 14:09:21] -> CREATE [C:\Users\yan20\Downloads\New Text Document.txt] (isDirectory: false)
CREATE C:\Users\yan20\Downloads\New Text Document.txt
[DEBUG] [2024-09-22 14:09:23] ENTRY_DELETE [C:\Users\yan20\Downloads\New Text Document.txt]
[DEBUG] [2024-09-22 14:09:23] -> DELETE [C:\Users\yan20\Downloads\New Text Document.txt] (isDirectory: false)
DELETE C:\Users\yan20\Downloads\New Text Document.txt
```

I just wasn't patient enough. That's it. There are so many directories in my download folder because I had executed some installation script in there which created them, that it had to register multiple thousand dirs before starting to watch them.

It worked with java 8 not because it was java 8, but because I wrote the issue comment above while running it, I can't believe it.

Okay, speaking of performance, what do you feel like is a "reasonable" amount of directories for this library to watch? I'm not an expert in file system event watching, this is the first time I'm doing something like this and I don't really know how these systems work.

So basically, you need to register yourself for every directory you want to get updates for first? Okay, that's good to know already.

**Comment by gmethvin** [comment]
Windows supports the `FILE_TREE` modifier for the underlying watch service, so directory-watcher itself only needs to register the root. The issue is likely that it's hashing all the files in the directory to get initial hashes. If you don't need the deduplication you can set `.fileHashing(false)`. Since this is usually the bottleneck, it's more dependent on the size of the files than the number of files/directories.
